packages/renderer/src/lib/image/ImageDetails.spec.ts

```diff
@@ -68,19 +68,22 @@ delete myNoneNameImage.RepoTags;
 const deleteImageMock = vi.fn();
 const hasAuthMock = vi.fn();
 
+beforeAll(() => {
+  Object.defineProperty(window, 'showMessageBox', { value: showMessageBoxMock });
+  Object.defineProperty(window, 'listImages', { value: listImagesMock });
+  Object.defineProperty(window, 'listContainers', { value: vi.fn() });
+  Object.defineProperty(window, 'deleteImage', { value: deleteImageMock });
+  Object.defineProperty(window, 'hasAuthconfigForImage', { value: hasAuthMock });
+  Object.defineProperty(window, 'getImageCheckerProviders', { value: vi.fn().mockResolvedValue([]) });
+  Object.defineProperty(window, 'listViewsContributions', { value: vi.fn().mockResolvedValue([]) });
+  Object.defineProperty(window, 'getImageFilesProviders', { value: vi.fn().mockResolvedValue([]) });
+  Object.defineProperty(window, 'getConfigurationProperties', { value: vi.fn().mockResolvedValue({}) });
+  Object.defineProperty(window, 'getContributedMenus', { value: getContributedMenusMock });
```

Comment:
question, why the multiple calls to `Object.defineProperty(window)` vs a single `Object.defineProperty(global, 'window', {`

Reply:
Because I translated all (window as any).xxx = to Object.property. We could merge but not sure it is better for readability